### PR TITLE
Small cleanups & colouring board squares

### PR DESCRIPTION
**Pull request by FoxLisk**

### CODE REVIEW

File: src/mind.rs [hunk 1]
```diff
@@ -82,11 +82,10 @@ fn order_moves(commits: &mut Vec<Commit>) {
 
 pub fn α_β_negamax_search(world: WorldState,
                           depth: u8,
-                          α: f32,
+                          mut α: f32,
```
Comment (FoxLisk) [inline src/mind.rs]: (this is the technique by which one acquires re-assignable parameter names)

File: src/life.rs [hunk 2]
```diff
@@ -891,16 +892,26 @@ impl WorldState {
 
 }
 
+fn next_color(color: Color) -> Color {
+    match color {
+        Color::Yellow => Color::Cyan,
+        Color::Cyan => Color::Yellow,
+        _ => unreachable!()
+    }
+}
```
Comment (zackmdavis) [inline src/life.rs]: I think there a [Cycle iterator](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.cycle) that we might want for this sort of thing?

```
zmd@ExpectedReturn:~/Code/Misc/Rust_Sandbox$ cat cycle_experiment.rs 
fn main() {
    let cycle_items = ["a", "b", "c"];
    for (index, item) in (0..10).zip(cycle_items.iter().cycle()) {
        println!("{}: {}", index, item);
    }
}
zmd@ExpectedReturn:~/Code/Misc/Rust_Sandbox$ rustc cycle_experiment.rs 
zmd@ExpectedReturn:~/Code/Misc/Rust_Sandbox$ ./cycle_experiment 
0: a
1: b
2: c
3: a
4: b
5: c
6: a
7: b
8: c
9: a
```

Comment (FoxLisk) [inline src/life.rs]: I couldn't figure out how to spell it... it was yelling at me about how Color doesn't have Clone defined on it. i can try again, tho.

**Comment by zackmdavis** [comment]
Hmm ...

unoccupied_ansi_tiles

I [think we could](http://zackmdavis.net/blog/2013/09/app-academy-diary-week-two/#wednesday) do better than this if we used [distinguishable](http://ogham.rustdocs.org/ansi_term/#extended-colours) [background colors](http://ogham.rustdocs.org/ansi_term/#bold,-underline,-and-background) rather than printing a \u2588 Full Block on onoccupied locales.

**Comment by FoxLisk** [comment]
you're right. however, that requires finding good colours to match with the orange and blue, and that sounds tedious and annoying :) this way at least I can see where my bishops are going.

**Comment by zackmdavis** [comment]
OK; I'll merge this now and pick background colors later.

**Pull request merged**